packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt

```diff
@@ -8,72 +8,68 @@ import java.io.IOException
-                }
-            }
-            "share", "shareWithResult" -> {
-                expectMapArguments(call)
-                if (isWithResult && !manager.setCallback(result)) return
+        if (isWithResult && !manager.setCallback(result)) return
 
-                // Android does not support showing the share sheet at a particular point on screen.
-                share.share(
-                    call.argument<Any>("text") as String,
-                    call.argument<Any>("subject") as String?,
-                    isWithResult,
-                )
+        try {
+            when (call.method) {
+                "shareUri" -> {
+                    share.share(
+                        call.argument<Any>("uri") as String, subject = null, withResult = false
+                    )
+                    success(isWithResult, isResultRequested, result)
+                }
 
-                if (!isWithResult) {
-                    if (isResultRequested) {
-                        result.success("dev.fluttercommunity.plus/share/unavailable")
-                    } else {
-                        result.success(null)
-                    }
+                "share", "shareWithResult" -> {
+                    share.share(
+                        call.argument<Any>("text") as String,
+                        call.argument<Any>("subject") as String?,
+                        isWithResult,
+                    )
+                    success(isWithResult, isResultRequested, result)
                 }
-            }
-            "shareFiles", "shareFilesWithResult" -> {
-                expectMapArguments(call)
-                if (isWithResult && !manager.setCallback(result)) return
 
-                // Android does not support showing the share sheet at a particular point on screen.
-                try {
+                "shareFiles", "shareFilesWithResult" -> {
                     share.shareFiles(
                         call.argument<List<String>>("paths")!!,
                         call.argument<List<String>?>("mimeTypes"),
                         call.argument<String?>("text"),
                         call.argument<String?>("subject"),
                         isWithResult,
                     )
-
-                    if (!isWithResult) {
-                        if (isResultRequested) {
-                            result.success("dev.fluttercommunity.plus/share/unavailable")
-                        } else {
-                            result.success(null)
-                        }
-                    }
-                } catch (e: IOException) {
-                    result.error("Share failed", e.message, null)
+                    success(isWithResult, isResultRequested, result)
                 }
+
+                else -> result.notImplemented()
+            }
+        } catch (e: IOException) {
+            manager.clear()
+            result.error("Share failed", e.message, null)
+        }
```

Comment:
Surround all the share calls with a try/catch that will call to `manager.clear()`, which fixes the actual bug, as well as return the error result.